.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,30 @@
+name: build-and-test
+on: [pull_request]
+jobs:
+  test-re2:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Checkout submodules
+        run: git submodule update --init --recursive
+      - name: Install arrow
+        run: |
+          sudo apt update
+          sudo apt install -y -V ca-certificates lsb-release wget
+          wget https://apache.bintray.com/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb
+          sudo apt install -y -V ./apache-arrow-archive-keyring-latest-$(lsb_release --codename --short).deb
+          sudo apt update
+          sudo apt install -y -V libarrow-dev
+      - name: Build dask_native
+        run: |
+          mkdir -p native/build
+          cd native/build
+          cmake ..
+          make
+          cd ../..
+      - name: Install python dependencies
```

Comment:
Take a look at the [setup-python](https://github.com/actions/setup-python) action. You can then use `python -m pip` or just `python`.